src/util/sss_krb5.c

```diff
@@ -245,23 +241,20 @@ errno_t select_principal_from_keytab(TALLOC_CTX *mem_ctx,
 
         ret = EOK;
     } else {
-        DEBUG(SSSDBG_MINOR_FAILURE, "No suitable principal found in keytab\n");
-        ret = ENOENT;
+        error_message = "No suitable principal found in keytab";
+        ret = EINVAL;
```

Comment:
Is there a reason you prefer EINVAL here? Currently the actual value of the error code is checked nowhere os it does not matter, but I still think ENOENT fits better.

Reply:
I understand initial intention, but `ENOENT == “No such file or directory”`  --  I think this is misleading as keytab file was found (and btw this error code was the root cause of initial bug report)

EINVAL == “Invalid argument”  --  might be not the best choice either, but I was unable to find anything more appropriate.

Would it be better to extend `utils_error.h:sssd_errors`?

Reply:
Yes, a dedicated error would be good.

Reply:
Done.